### PR TITLE
fix: support modal + drawer combo; fix drawer focus trap

### PR DESCRIPTION
**Pull request by cjhensen**
## Summary

**Drawers**
The focus trap in all drawers is currently broken - you can tab out into the browser.
The fix for this was to render the portal separately from the drawer, in the same fashion as the modal component. The cause is still not 100% clear, but is likely due to some combination of context or re-rendering issues.

**Drawer + Modal combo** 
When a modal is triggered from a button inside a drawer, the event handlers become funky as both components are listening for the same Escape key event.
The fix: add/remove the event listeners based on if focus is within the drawer or modal.

**Screen reader accessibility for drawers and modals**
Changed aria-labelledby to aria-describedby so screen readers can properly announce the modal title

### CODE REVIEW

File: packages/drawer/src/Drawer.tsx [hunk 1]
```diff
@@ -45,7 +46,13 @@ type DrawerProps = {
   hideCancel?: boolean;
 };
 
-const Drawer = ({
+const Drawer = (props: DrawerProps) => (
+  <Portal>
+    <DrawerContainer {...props} />
+  </Portal>
+);
```
Comment (cjhensen) [inline packages/drawer/src/Drawer.tsx:53]: rendering the portal separately from the drawer contents fixed the issue of the broken focus trap

File: packages/drawer/src/Drawer.tsx [hunk 2]
```diff
@@ -55,30 +62,48 @@ const Drawer = ({
...
+      removeOverlayAndEventHandler();
+    }
 
     return () => {
-      document.body.classList.remove('has-overlay');
-      document.removeEventListener('keydown', handleEscape);
+      removeOverlayAndEventHandler();
     };
-  }, [onCancel, testId]);
+  }, [onCancel, testId, isFocusWithin]);
```
Comment (cjhensen) [inline packages/drawer/src/Drawer.tsx:106]: this is what allows for smoother drawer + modal focus behavior. adding/removing event listeners based on if the element is focused within

**Review by sunnyguduru — APPROVED** [review]
I don't have much context into drawers but the modal code looks good

**Review by apucacao — APPROVED** [review]
This make sense.

I wonder if Modal and Drawer should share some of their "modality" in our implementation: they're so similar.

**Pull request merged**